solutions/src/main/java/ru/sberbank/school/task13/cache/CacheProxy.java

```diff
@@ -0,0 +1,334 @@
+
+            Object result = null;
+            Cache cacheAnnotation = method.getAnnotation(Cache.class);
+
+            if (!handlerInvokeMethodArgsCheck(cacheAnnotation, method, args)) {
+                return method.invoke(service, args);
+            }
+
+            CacheType cacheType = cacheAnnotation.cacheType();
+
+            if (cacheType.equals(CacheType.JVM)) {
+                result = invokeJVM(method, args);
+            } else if (cacheType.equals(CacheType.FILE)) {
+                result = invokeFile(method, args);
+            } else {
+                throw new CacheProxyException("unknown Cache type is declared in " + method.getName());
+            }
+
+            return result;
+        }
+
+        private boolean handlerInvokeMethodArgsCheck(Annotation cacheAnnotation, Method method, Object[] args) {
+            if (cacheAnnotation == null) {
+                return false;
+            }
+
+            if (method.getReturnType().equals(void.class)) {
+                throw new IllegalArgumentException("called method can not be void");
+            }
+
+            return true;
+        }
+
+        private Object invokeJVM(Method method, Object[] args) throws InvocationTargetException,
+                IllegalAccessException {
+
+            String keyFromInput = getKeyToCache(method, args);
+
+            Object result = findInCache(keyFromInput);
+
+            if (result == null) {
+                result = method.invoke(service, args);
+                cacheMap.put(keyFromInput, result);
+            }
+
+            return result;
+        }
+
+        private Object invokeFile(Method method, Object[] args) throws InvocationTargetException, IllegalAccessException {
+            String key = getKeyToCache(method, args);
+            String fileName = key + ".cache";
+
+            File file = new File(directoryPath + File.separator + fileName);
+            Object result = null;
+            File fileInCache = findInDirectory(file);
+
+            if (fileInCache != null) {
+                result = loadFromCacheFile(fileInCache);
+            } else {
+                result = method.invoke(service, args);
+                writeToCacheFile(file, result);
+            }
+
+            return result;
+        }
+
+        private Object findInCache(String toFind) {
+            Object result = null;
+
+            if (cacheMap.containsKey(toFind)) {
+                result = cacheMap.get(toFind);
+            } else {
+                for (String keyFromCacheMap : cacheMap.keySet()) {
+                    if (keyFromCacheMap.contains(toFind)) {
+                        result = cacheMap.get(keyFromCacheMap);
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        private File findInDirectory(File toFind) {
+            String fullNameToFind = isZip(toFind.getAbsolutePath()) ? (toFind.getAbsolutePath() + ".zip") : toFind.getAbsolutePath();
```

Comment:
Line is longer than 120 characters (found 133).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1734472260>*